cmd/gateway_validate.go

```diff
@@ -210,6 +211,24 @@ this command unless --online flag is used.
 			if len(validateCmdKongStateFile) == 0 {
 				validateCmdKongStateFile = []string{"-"}
 			}
+
+			// Iterate over the input values and validate them against the keys in entityMap
+			for _, value := range validateCmdOnlineEntitiesFilter {
+				// Check if the value is valid by comparing it with keys in entityMap
+				isValid := false
+				for key := range validate.EntityMap {
+					if value == key {
+						isValid = true
+						break
+					}
+				}
+
+				if !isValid {
+					// Generate an error message stating that the value should be part of Kong entities
+					return fmt.Errorf("invalid value '%s' for --check-online-plugins-only; it should be a valid Kong entity",
```

Comment:
Can we return the valid keys from `EntityMap` here? Don't just tell them that something is invalid - tell them what the possible values are

Reply:
Yes good idea, here is the new err message:

`Error: invalid value 'Routess' for --check-online-plugins-only; it should be a valid Kong entity. Valid entities: [KeyAuths RBACEndpointPermissions Plugins CACertificates Certificates JWTAuths Oauth2Creds SNIs RBACRoles Services ACLGroups Consumers HMACAuths Routes Upstreams BasicAuths Documents Targets FilterChains Vaults]`

<img width="1001" alt="Screenshot 2024-12-02 at 14 29 23" src="https://github.com/user-attachments/assets/4d994f66-e6b0-4c86-b272-7adceede64a1">